src/components/Articlelist.jsx

```diff
@@ -15,6 +16,16 @@ export const Articlelist = () => {
             })
     }, [])
 
+    useEffect(() => {
+        getArticles(undefined, topic.topic)
+        .then((articles) => {
+            setArticles(articles);
+            Isloading(false)
+        })
+    }, [topic])
+
+
```

Comment:
You can definitely dry this component up. 

Two useEffects aren't necessary because axios should be able to handle if topic is not defined. You should be able to just include the second useEffect.

The second problem you're probably coming up against is that you've got author in your API function messing up the param order and therefore requiring you to put undefined in here. I'd strongly advise you don't add code in here that you don't need yet. AKA authors, sort by and order by can all disappear for the time being from this API call and you can focus solely on topic.